packages/frontend/src/components/DownloadCsvPopup/index.tsx

```diff
@@ -108,33 +92,21 @@ const DownloadCsvPopup: FC<Props> = memo(({ fileName, rows, getCsvLink }) => {
                             />
                         </InputWrapper>
                     )}
-
-                    <CSVExporter
-                        linkRef={csvRef}
-                        data={csvRows}
-                        filename={csvFilename}
-                    />
-
                     <Button
                         fill
                         intent={Intent.PRIMARY}
                         icon="export"
                         onClick={() => {
-                            if (limit === Limit.TABLE) {
-                                setCsvRows(
-                                    getResultValues(
-                                        rows,
-                                        format === Values.RAW,
-                                    ),
-                                );
-                            } else {
-                                getCsvLink(
-                                    limit === Limit.CUSTOM ? customLimit : null,
-                                    format === Values.RAW,
-                                ).then((url) => {
-                                    window.open(url, '_blank');
-                                });
-                            }
+                            getCsvLink(
```

Comment:
so are you removing the csv from cached results and always downloading the csv from the new file ? 

I guess it makes sense to unify the code, it's a shame we can't reuse the data from the query we have...

Reply:
we'll probably have to find another solution for the chart in explore view cause this method can't be applied